<launch>
  <!-- launch video stream -->
  <include file="$(find video_stream_opencv)/launch/camera.launch" >
        <!-- node name and ros graph name -->
        <arg name="camera_name" value="webcam" />
        <!-- means video device 0, /dev/video0 -->
        <arg name="video_stream_provider" value="0" />
        <!-- throttling the querying of frames to -->
        <arg name="fps" value="30" />
        <!-- setting frame_id -->
        <arg name="frame_id" value="webcam" />
        <!-- camera info loading, take care as it needs the "file:///" at the start , e.g.:
        "file:///$(find your_camera_package)/config/your_camera.yaml" -->
        <arg name="camera_info_url" value="file:///$(find robo_magellan)/config/logitech-C270.yaml" />
        <!-- flip the image horizontally (mirror it) -->
        <arg name="flip_horizontal" value="false" />
        <!-- flip the image vertically -->
        <arg name="flip_vertical" value="false" />
        <!-- visualize on an image_view window the stream generated -->
        <arg name="visualize" value="true" />
  </include>

  <node name="cone_finder" pkg="robo_magellan" type="detect_cones.py">
    <rosparam command="load" file="$(find robo_magellan)/launch/cone_finder.yaml" />
    <param name="~captureVideo" value="False" />
    <param name="~publishImages" value="True" />
    <param name="~ignorePriorDetections" value="True" />

	<!-- Use 'bin' for new binning algorithm, 'hsv' for hue-based algorithm. -->
    <param name="~minConeArea" value="300" />
	<param name="~thresholdAlgorithm" value="bin" />
	<param name="~binConfig" value="$(find robo_magellan)/config/realsense-bins-0_6.json" />
    <param name="~contourFilterAlgorithm" value="convexHull" />
    <remap from="/camera/color/camera_info" to="/webcam/camera_info" />
	<remap from="/camera/color/image_raw" to="/webcam/image_raw" />
  </node>
  <node name="image_view" pkg="image_view" type="image_view" args="image:=/cone_finder/colorImage" output="screen" />
</launch>
